src/sonic-config-engine/portconfig.py

```diff
@@ -45,16 +46,23 @@
 #
 # Helper Functions
 #
+
+# For python2 compatibility
```

Comment:
I made changes addressing your comments
1. on non-ascii chars - I kept the behavior the same as previous - instead of throwing UnicodeEcodeError, encode with hex escapes.
2. moved the compatibility code under sys.version_info.major == 2 check
3. added bool values to sample_platform.json. The changes are already covered by test_cfggen_platformJson

Reply:
@qiluo-msft please review

Reply:
@qiluo-msft please review

Reply:
@qiluo-msft @saiarcot895 thanks